index.js

```diff
@@ -0,0 +1,125 @@
+const URL = 'https://trektravel.herokuapp.com/trips';
+
+const reportStatus = (message) => {
+  $('#status-message').html(message);
+};
+
+const reportError = (message, errors) => {
+  let content = `<p>${message}</p>`
+  content += "<ul>";
+  for (const field in errors) {
+    for (const problem of errors[field]) {
+      content += `<li>${field}: ${problem}</li>`;
+    }
+  }
+  content += "</ul>";
+  reportStatus(content);
+};
+
+const loadTrips = () => {
+  reportStatus('Loading trips...');
+  
+  axios.get(URL)
+    .then((response) => {
+      reportStatus(`Successfully loaded ${response.data.length} trips`);
+      const trips = response.data;
+      displayTripList(trips);
+    })
+    .catch((error) => {
+      reportStatus(`Encountered an error while loading trips: ${error.message}`);
+      console.log(error);
+    });
+};
+
+const displayTripList = (tripList) => {
+  $(".card").show();
+  const target = $('#trip-list');
+  target.empty();
+  tripList.forEach((trip) => {
+    target.append(`<a href="#" id="${trip.id}" class="list-group-item list-group-item-action">${trip.name}</a>`);
+    const getDetails = showTripDetails(trip)
+    $(`#${trip.id}`).click(getDetails);
+  });
+}
+
+const showTripDetails = (trip) => {
+  console.log("showing details for trip", trip.name);
+
+  const getTripDetails = () => {
+    axios.get(URL + '/' + trip.id)
+    .then((response) => {
+      const tripInfo = response.data;
+      reportStatus(`Successfully loaded trip ${tripInfo.name}`);
+      displayTripDetails(tripInfo)
+      const reserve = reserveTrip(trip)
+      const resHeader = $('#reservation-header')
+      resHeader.text(`Reserve a Spot on ${tripInfo.name}`)
+      const target = $('#reservation-form');
+      target.removeClass('d-none');
+      $('#reservation-card').removeClass('d-none');
+      resHeader.removeClass('d-none');
+      target.submit(reserve)
```

Comment:
When reserving a second trip, your site makes two POST requests. To reproduce:
1. Open page
1. Open chrome dev tools to the "Network" tab
1. Click See All Trips
1. Click on the first trip (Cairo to Zanzibar)
1. Make a reservation and submit
1. Click on the second trip (Everest Base Camp Trek)
1. Make a reservation and submit
1. Observe that on submit, 2 POST requests were made to trek API's /reservations endpoint

The reason for this is that you reuse the same reservation form for each trip, but re-attach the `submit` handler every time you show details for a trip.